op/model.py

```diff
@@ -237,6 +294,63 @@ def __init__(self, backend):
     def _load(self):
         return self._backend.config_get()
 
+class Status(ABC):
+    """Status values specific to applications and units."""
+
+    _statuses = {}
+
+    def __init__(self, message):
+        self.message = message
+
+    def __new__(cls, *args, **kwargs):
+        if cls is Status:
+            raise TypeError("cannot instantiate a base class")
+
+        return super().__new__(cls)
+
+    @classmethod
+    def _register_status(cls, name, type_):
+        """For use by subclasses only."""
+        cls._statuses[name] = type_
+
+    @classmethod
+    def from_string(cls, name, message):
```

Comment:
s/_string/_name/

... since we call the attribute _name._

Reply:
OK, makes sense.